models/task.rb

```diff
@@ -0,0 +1,182 @@
+      end
+    rescue PG::InvalidTextRepresentation => e   
+      puts "Invalid Status Entered .... "
+      puts e.message
+    rescue PG::SyntaxError => e
+      puts 'Error: A syntax error occurred in the SQL query.'
+      puts e.message
+    rescue PG::ConnectionBad => e
+      puts 'Error: The database connection is invalid or closed.'
+      puts e.message 
+    rescue ArgumentError => e
+      puts "Error parsing due_date: #{e.message}"
+    rescue TypeError => e
+      puts "Error formatting due_date: #{e.message}"    
+    end
+  end
+
+  def self.find_by_id(task_id)
+    begin 
+      result = @@connection.exec_params("SELECT * FROM tasks WHERE id=$1", [task_id])
+      if result.ntuples.zero?
+        puts "No Task Found"
+      else
+        result = result[0].transform_keys(&:to_sym)
+        new(
+          id: task_id,
+          assignee_user: result[:assignee_user_id],
+          description: result[:description],
+          due_date: Date.parse(result[:due_date]),
+          priority: result[:priority],
+          creator: result[:creator_id],
+          status: result[:status],
+        )
+      end
+    rescue PG::SyntaxError => e
+      puts 'Error: A syntax error occurred in the SQL query.'
+      puts e.message
+    rescue PG::ConnectionBad => e
+      puts 'Error: The database connection is invalid or closed.'
+      puts e.message   
+    end
+  end
+
+  def self.next_task(user)
+    begin
+      result = @@connection.exec_params("SELECT * FROM tasks WHERE assignee_user_id = $1 ORDER BY priority DESC LIMIT 1", [user.id])
+      if result.ntuples.zero?
+        puts "No Task Found"
+      else
+        result = result[0].transform_keys(&:to_sym)
+        new(
+          id: result[:id],
+          assignee_user: result[:assignee_user_id],
+          description: result[:description],
+          due_date: Date.parse(result[:due_date]),
+          priority: result[:priority],
+          creator: result[:creator_id],
+          status: result[:status],
+        )
+      end
+    rescue PG::SyntaxError => e
+      puts 'Error: A syntax error occurred in the SQL query.'
+      puts e.message
+    rescue PG::ConnectionBad => e
+      puts 'Error: The database connection is invalid or closed.'
+      puts e.message 
+    end
+  end
+
+  def self.create_task(assignee_user:, description:, due_date:, priority:, creator_email:, status:)
+    begin
+      date = Date.parse(due_date)
+      formatted_due_date = Date.parse(due_date).strftime('%Y-%m-%d')
+      user = User.find_by_email(creator_email)
+      new(
+        assignee_user: assignee_user,
+        description: description,
+        due_date: formatted_due_date,
+        priority: priority,
+        creator: user.id,
+        status: status
+      )
+    rescue ArgumentError => e
+      puts "Error parsing due_date: #{e.message}"
+    rescue TypeError => e
+      puts "Error formatting due_date: #{e.message}"  
+    end
+  end
+
+  def self.show_all_tasks
```

Comment:
```suggestion
  def self.show_all
```
we are already inside Task class, show_all should not have a suffix with `task`, it is implicit